# Added fix to make the tables and charts utilize available screen size, added some other fixes.

damiadek opened the pull request:
**It fixes issue #109** 

--Made the table adjust to data width and the chart to expand to fill available page space on all pages.
--Switched position of tables and charts in selector on all pages.
--Made chart default view for all pages with chart view.
--Changed order of menu items

![image](https://user-images.githubusercontent.com/15652001/62086811-7c5ed780-b257-11e9-81cd-b9ddfcf014ab.png)
![image](https://user-images.githubusercontent.com/15652001/62086843-939dc500-b257-11e9-9cbb-72f348b178e8.png)

![image](https://user-images.githubusercontent.com/15652001/62091274-0747ce00-b268-11e9-9b44-f3ebff046cc4.png)

![image](https://user-images.githubusercontent.com/15652001/62091322-2e060480-b268-11e9-8a70-356ccdcc5ccf.png)

![image](https://user-images.githubusercontent.com/15652001/62091344-4413c500-b268-11e9-88a8-ef54b4395e28.png)



--Made 1D selected by default for interval in exchanges view.

![image](https://user-images.githubusercontent.com/15652001/62086901-be881900-b257-11e9-82d6-47290452e3f9.png)


--Fixed typo in sample-config
![image](https://user-images.githubusercontent.com/15652001/62087005-1030a380-b258-11e9-92c8-14e33a199105.png)

![image](https://user-images.githubusercontent.com/15652001/62091350-52fa7780-b268-11e9-894e-87d37f90841b.png)

damiadek commented:
Fixes issue #109

damiadek commented:
Fixes issue #109. Added fix to make the tables and charts utilize available screen size.

raedah commented:
The table should not be being force to take 100% of available width. It should only use the needed width. It should not have an artificially limited width.

damiadek commented:
![Capture](https://user-images.githubusercontent.com/15652001/62000865-17ca3e00-b0da-11e9-9ed7-1c64b04f67c4.PNG)
![image](https://user-images.githubusercontent.com/15652001/62000868-2c0e3b00-b0da-11e9-9cf2-f066cd87fcd1.png)

damiadek commented:
Switched position of tables and charts control on all pages.

Made chart default view for all pages with chart view

Made 1D selected by default for interval in exchanges view

Changed order of menu items 

![image](https://user-images.githubusercontent.com/15652001/62015461-b79fce80-b1a3-11e9-9548-aa378e7321ca.png)

Sirmorrison commented on web/public/app/src/controllers/exchange_controller.js:
```diff
@@ -23,11 +23,11 @@ export default class extends Controller {
     this.selectedCurrencyPairTarget.value = cpair[0].text
     this.selectedNumTarget.value = num[0].text
-    this.selectedIntervalTarget.value = interval[0].value
+    this.selectedIntervalTarget.value = interval[4].value
```
Why did you make 1d the default view?

damiadek replied:
It was requested by @raedah

Sirmorrison commented on web/public/app/src/css/style.scss:
```diff
@@ -400,7 +405,7 @@ table.collapsible .labels tr td {
   padding: 0, 0, 0, 1rem !important;
   white-space: nowrap;
-  display: inline-block;
+  // display: inline-block;
```
remove commented line of code.

Sirmorrison commented on web/public/app/src/controllers/exchange_controller.js:
```diff
@@ -23,11 +23,11 @@ export default class extends Controller {
 
   initialize () {
-    this.viewOption = 'table'
+    this.setChart()
```
This would give you some major issues as you are calling up  the chart view function on initialization of the controller. If you where to call a function it should be the table function. Reason is the table view is the default view.

Again what if we choose to rename the view function from `setChart` to maybe `displayChartView`. It means we will have to rename if for every instance where it is initialize as compared to initializing `this.viewOption` where all a dev need is to choose to rename either `table` or `chart`.

so i think the change really isn't necessary.

damiadek replied:
The chart loading by default was also requested by @raedah . It's my bad, let me add the tasks he added to the description.


Also, setting `this.viewOption` was my first option, but, it did not have the required effect.

damiadek replied:
I think it's better to reference a function than to use a string literal to modify the view option. It would give you the flexibility to switch your `setChart` and `setTable` functions incase functionalities change.

Sirmorrison commented on web/public/app/src/controllers/mempool_controller.js:
```diff
@@ -26,7 +26,7 @@ export default class extends Controller {
     }
     this.dataType = 'size'
-    this.viewOption = 'table'
+    this.setChart()
```
Refer to my comment above.

Sirmorrison commented on web/public/app/src/controllers/pow_controller.js:
```diff
@@ -15,7 +15,7 @@ export default class extends Controller {
 
   initialize () {
-    this.viewOption = 'table'
+    this.setChart()
```
Same as above.

Sirmorrison commented on web/public/app/src/controllers/vsp_controller.js:
```diff
@@ -28,14 +28,14 @@ export default class extends Controller {
         this.vsps.push(chartSource.value)
       }
     })
+    this.setChart()
```
Same as above.

Sirmorrison commented on web/views/error.html:
```diff
@@ -1,15 +1,16 @@
+            <div class="container-fluid text-center">
+                <h3 class="text-danger mb-3">Oops</h3>
+                <div class="alert alert-danger">{{ .error }}</div>
+                <p>Refresh or <a href="/">Go Home</a></p>
```
Change `Go home` to `Return home`

Sirmorrison commented on web/views/error.html:
```diff
@@ -1,15 +1,16 @@
         </div>
     </div>
-</div>
 </body>
```
add an empty line after the `</body>` tag

Sirmorrison commented on web/views/exchange.html:
```diff
@@ -161,5 +170,6 @@
-    </html>
\ No newline at end of file
+</body>
+
```
remove empty space.

Sirmorrison commented on web/views/exchange.html:
```diff
@@ -161,5 +170,6 @@
\ No newline at end of file
+</body>
+
+</html>
```
add an empty space after `</html>` tag

Sirmorrison commented on web/views/home.html:
```diff
@@ -1,14 +1,16 @@
     </div>
     {{ template "footer" }}
 </body>
+
```
remove empty space and add it after `</html>` tag

Sirmorrison commented on web/views/layout.html:
```diff
@@ -8,42 +9,41 @@
-                <ul class="navbar-nav mr-auto">
+                <ul class="navbar-nav mx-auto">
                     <li class="nav-item">
-                        <a class="nav-link" id="nav-exchanges" href="/exchanges">
```
I noticed you re-ordered the nav items was it based on instruction from @raedah or ?

damiadek replied:
Based on instruction

Sirmorrison commented on web/views/mempool.html:
```diff
@@ -156,4 +149,5 @@ <h4>Mempool</h4>
 </body>
-</html>
+
+</html>
```
empty line.

Sirmorrison commented on web/views/pow.html:
```diff
@@ -1,123 +1,120 @@
-{{ template "footer" }}
+    {{ template "footer" }}
 </body>
+
```
remove empty line.

Sirmorrison commented on web/views/vsp.html:
```diff
@@ -1,172 +1,179 @@
                 <div class="control-wrapper">
-                    <div class="chart-control-wrapper mb-2" data-target="vsp.chartSelector">
-                        <div class="chart-control">
+                    <!-- <div class="chart-control-wrapper mb-2" data-target="vsp.chartSelector">
```
remove commented lines of codes.

Sirmorrison commented:
![image](https://user-images.githubusercontent.com/27733432/62129848-95a36a80-b2cf-11e9-97ce-01847fe261b4.png)

i observed that the tables are all either center aligned or occupying the entire width of the screen. Please centralize this table.

damiadek commented:
@Sirmorrison ,



**--- Fixed.**
> i observed that the tables are all either center aligned or occupying the entire width of the screen. Please centralize this table.
![image](https://user-images.githubusercontent.com/15652001/62167231-ad541080-b31a-11e9-8b2a-4c73054f9c6b.png)



**--- Fixed.**
> PoW is not displaying any data please fix it.
![image](https://user-images.githubusercontent.com/15652001/62167269-bf35b380-b31a-11e9-8413-8239d07d4abb.png)



**--- This happens when the screens size is too small.**
> fix the issue with the warp on the view selector.


**--- Fixed.**
> and since everything is centralized now, i think it will be nice for every thing on the selector section to be equally spaced out.


**--- Which table has fixed width?.**
> I also noticed the tables have fixed width. From the description f what was fixed was for the tables and chart to utilize available space.

Sirmorrison commented:
![image](https://user-images.githubusercontent.com/27733432/62170115-33c02080-b322-11e9-9600-6d9409bdd8a0.png)

fix issue with the pagination wrap.. this was fixed some time back. the pagination should remain inline.

Sirmorrison commented on web/views/exchange.html:
```diff
@@ -1,103 +1,112 @@
-                        <div class="chart-control">
+                    <div class="d-flex flex-row bottom-ctl">
+                        <div class="chart-control ml-auto mr-2 my-2">
                             <ul class="nav nav-pills">
```
add `style="white-space: nowrap;`

Sirmorrison commented on web/public/app/src/controllers/pow_controller.js:
```diff
@@ -128,8 +128,9 @@ export default class extends Controller {
     pows.forEach(pow => {
-      data.push(new Date(pow.Time))
-      data.push(pow.pool_hashrate_th)
+      pow.time = this.formatPowDateTime(pow.time)
```
Revert this to what it use to be.

Sirmorrison commented on web/public/app/src/controllers/pow_controller.js:
```diff
@@ -160,4 +161,10 @@ export default class extends Controller {
     })
   }
+
+  formatPowDateTime (dateTime) {
```
Remove this function.
